unchanged:
--- a/arch/arm64/boot/dts/allwinner/Makefile
+++ b/arch/arm64/boot/dts/allwinner/Makefile
@@ -23,10 +23,11 @@ dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-pinetab-early-adopter.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-sopine-baseboard.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-recore.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-recore-a5.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-recore-a6.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-recore-a7.dtb
+dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-recore-a8.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a64-teres-i.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-a100-allwinner-perf1.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-bananapi-m2-plus.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-bananapi-m2-plus-v1.2.dtb
 dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h5-emlid-neutis-n5-devboard.dtb
unchanged:
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a5.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a5.dts
@@ -55,11 +55,48 @@ &reg_dldo4 {
 	regulator-min-microvolt = <700000>;
 	regulator-max-microvolt = <700000>;
 	regulator-name = "vcc-vref-e3";
 };
 
+&reg_dcdc5 {
+	regulator-min-microvolt = <1500000>;
+	regulator-max-microvolt = <1500000>;
+};
+
 /* STM32 */
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart4_pins>;
 	status = "okay";
-};
\ No newline at end of file
+};
+
+&mali {
+	/delete-property/ operating-points-v2;
+	assigned-clock-rates = <120000000>;
+};
+
+&cpu0 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu1 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu2 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu3 {
+	/delete-property/ cpu-idle-states;
+};
+
+/ {
+	cpus {
+		/delete-node/ idle-states;
+	};
+	/delete-node/ scpi;
+};
+
+&mbus {
+	status = "disabled";
+};
unchanged:
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a6.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a6.dts
@@ -18,5 +18,42 @@ &reg_aldo1 {
 &uart4 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart4_pins>;
 	status = "okay";
 };
+
+&reg_dcdc5 {
+	regulator-min-microvolt = <1500000>;
+	regulator-max-microvolt = <1500000>;
+};
+
+&mali {
+	/delete-property/ operating-points-v2;
+	assigned-clock-rates = <120000000>;
+};
+
+&cpu0 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu1 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu2 {
+	/delete-property/ cpu-idle-states;
+};
+
+&cpu3 {
+	/delete-property/ cpu-idle-states;
+};
+
+/ {
+	cpus {
+		/delete-node/ idle-states;
+	};
+	/delete-node/ scpi;
+};
+
+&mbus {
+	status = "disabled";
+};
unchanged:
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a7.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a7.dts
@@ -6,10 +6,22 @@
 #include <dt-bindings/usb/pd.h>
 
 / {
 	model = "Recore-A7";
 	compatible = "iagent,recore-a7", "allwinner,sun50i-a64";
+
+	reg_usb0_vbus: usb0-vbus {
+		compatible = "regulator-fixed";
+		vin-supply = <&reg_5v>;
+		regulator-name = "usb0-vbus0";
+		regulator-type = "voltage";
+		gpio = <&pio 3 7 GPIO_ACTIVE_HIGH>;
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		status = "okay";
+	};
 };
 
 &reg_usb1_vbus3 {
 	gpio = <&pio 3 5 GPIO_ACTIVE_HIGH> ; /* PD5 */
 };
@@ -28,14 +40,15 @@ &reg_aldo1 {
 /* Set over current to lowest value: 35 A = 700 mV */
 &reg_dldo2 {
 	regulator-min-microvolt = <700000>;
 	regulator-max-microvolt = <700000>;
 };
-
+	
 &usbphy {
-	usb0_id_det-gpios   = <&pio 4 7 GPIO_ACTIVE_LOW>;
+	usb1_vbus-supply = <&reg_5v>;
 };
+
 &uart2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart2_pins>;
 	status = "okay";
 };
unchanged:
--- /dev/null
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore-a8.dts
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+
+/dts-v1/;
+
+#include "sun50i-a64-recore.dtsi"
+#include <dt-bindings/usb/pd.h>
+
+/ {
+	model = "Recore-A8";
+	compatible = "iagent,recore-a8", "allwinner,sun50i-a64";
+
+	reg_usb0_vbus: usb0-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb0-vbus0";
+		regulator-type = "voltage";
+		gpio = <&pio 3 7 GPIO_ACTIVE_HIGH>; // PD7: EN-USB-OTG
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		enable-active-high;
+		status = "okay";
+	};
+};
+
+&pio{
+	reg_usb1_vbus4_pins: reg-usb1-vbus4-pins {
+		pins = "PD5";
+		function = "reg_usb1_vbus4";
+	};
+	reg_usb1_vbus3_pins: reg-usb1-vbus3-pins {
+		pins = "PD6";
+		function = "reg_usb1_vbus3";
+	};
+	reg_usb0_vbus_pins: reg-usb0-vbus-pins {
+		pins = "PD7";
+		function = "reg_usb0_vbus";
+	};
+	fusb0_int_pin: fusb0-int-pin {
+		pins = "PH10";
+		function = "gpio_in";
+	};
+};
+
+&reg_usb1_vbus3 {
+	gpio = <&pio 3 5 GPIO_ACTIVE_HIGH>;
+};
+
+&reg_usb1_vbus4 {
+	gpio = <&pio 3 6 GPIO_ACTIVE_HIGH>;
+};
+
+&reg_aldo1 {
+	regulator-always-on;
+	regulator-min-microvolt = <3300000>;
+	regulator-max-microvolt = <3300000>;
+	regulator-name = "vcc-ext";
+};
+
+/* Set over current to lowest value: 30 A = 1200 mV */
+&reg_dldo2 {
+	regulator-min-microvolt = <1200000>;
+	regulator-max-microvolt = <1200000>;
+};
+
+/* stm32 mcu */
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart2_pins>;
+	status = "okay";
+};
+
+&usb_otg {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&usb_power_supply {
+	regulator-name = "usb-power-supply";
+	status = "okay";
+};
+
+
+&usbphy {
+	status = "okay";
+	usb0_id_det-gpios = <&pio 4 7 GPIO_ACTIVE_HIGH>;
+    usb0_vbus_det-gpios = <&pio 4 6 GPIO_ACTIVE_HIGH>;
+	/*usb0_vbus_power-supply = <&usb_power_supply>;*/
+	/*usb0_vbus-supply = <&reg_usb0_vbus>;*/
+	/*usb1_vbus-supply = <&reg_5v>;*/
+	ports {
+		port@0 {
+			usb0_role_switch: endpoint {
+				remote-endpoint = <&fusb302_ep>;
+			};
+		};
+	};
+};
+
+
+
+&i2c1 {
+	status = "okay";
+
+	fusb0: fusb30x@22 {
+		compatible = "fcs,fusb302";
+		reg = <0x22>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&fusb0_int_pin>;
+		vbus-supply = <&reg_usb0_vbus>;
+		interrupt-parent = <&pio>;
+		interrupts = <7 10 IRQ_TYPE_EDGE_FALLING>; // PH10
+
+		port {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			usb_hs: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&usb0_hs>;
+			};
+		};
+
+		usb_con: connector {
+			compatible = "usb-c-connector";
+			data-role = "dual";
+			label = "USB-C";
+			op-sink-microwatt = <1000000>;
+			power-role = "dual";
+			sink-pdos =	<PDO_FIXED(5000, 1000, PDO_FIXED_USB_COMM)>;
+			source-pdos =	<PDO_FIXED(5000, 1000, PDO_FIXED_USB_COMM)>;
+			try-power-role = "sink";
+
+			ports {
+				port@0 {
+					fusb302_ep: endpoint {
+						remote-endpoint = <&usb0_role_switch>;
+					};
+				};
+			};
+		};
+	};
+};
+
+
+&ohci0 {
+	status = "okay";
+};
+
+&ehci0 {
+	status = "okay";
+	port {
+		usb0_hs: endpoint {
+			remote-endpoint = <&usb_hs>;
+		};
+	};
+};
\ No newline at end of file
only in patch2:
unchanged:
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-recore.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore.dts
@@ -27,5 +27,11 @@ reg_usb1_vbus6: usb1-vbus6 {
    	regulator-always-on;
    	enable-active-high;
    	vin-supply = <&reg_5v>;
    };
 };
+
+&mali {
+	/delete-property/ operating-points-v2;
+	assigned-clock-rates = <120000000>;
+};
+	
\ No newline at end of file
only in patch2:
unchanged:
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-recore.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-recore.dtsi
@@ -26,10 +26,11 @@ port {
 			hdmi_con_in: endpoint {
 				remote-endpoint = <&hdmi_out_con>;
 			};
 		};
 	};
+	
 	leds {
 		compatible = "gpio-leds";
 
 		heartbeat-led {
 			label = "recore:white:heartbeat";
@@ -40,11 +41,11 @@ heartbeat-led {
 		};
 
 		emmc-led {
 			label = "recore:white:emmc";
 			gpios = <&pio 1 5 GPIO_ACTIVE_HIGH>;
-			linux,default-trigger = "mmc0";
+			linux,default-trigger = "mmc2";
 		};
 
 		cpu-led {
 			label = "recore:white:cpu";
 			gpios = <&pio 1 6 GPIO_ACTIVE_HIGH>;
@@ -187,25 +188,19 @@ &mmc2 {
 	vqmmc-supply = <&reg_eldo1>;
 	non-removable;
 	bus-width = <8>;
 	cap-mmc-hw-reset;
 	mmc-hs200-1_8v;
+	post-power-on-delay-ms = <1>; /* power is already turned on by the bootloader */
 	status = "okay";
 };
 
-&ohci0 {
-	status = "okay";
-};
 
 &ohci1 {
 	status = "okay";
 };
 
-&ehci0 {
-	status = "okay";
-};
-
 &ehci1 {
 	status = "okay";
 };
 
 &pio {
@@ -294,13 +289,13 @@ &reg_dldo2 {
 	regulator-max-microvolt = <1000000>;
 	regulator-name = "current-control";
 };
 
 &reg_eldo1 {
-	regulator-always-on;
+	regulator-boot-on;
 	regulator-min-microvolt = <1800000>;
-	regulator-max-microvolt = <1800000>;
+	regulator-max-microvolt = <3300000>;
 	regulator-name = "emmc-io";
 };
 
 &reg_fldo1 {
 	regulator-name = "vdd-hsic";
@@ -312,15 +307,15 @@ &reg_fldo2 {
 	regulator-min-microvolt = <1100000>;
 	regulator-max-microvolt = <1100000>;
 	regulator-name = "vdd-cpus";
 };
 
-&gpio0_ldo {
+&reg_ldo_io0 {
 	function = "ldo";
 };
 
-&gpio1_ldo {
+&reg_ldo_io1 {
 	function = "ldo";
 };
 
 &reg_ldo_io0 {
 	regulator-always-on;
@@ -369,39 +364,7 @@ &usb_otg {
 &usb_power_supply {
 	status = "okay";
 };
 
 &usbphy {
-	usb0_vbus_power-supply = <&usb_power_supply>;
-	vcc-supply = <&reg_dcdc1>;
 	status = "okay";
 };
-&cpu0 {
-	/delete-property/ cpu-idle-states;
-};
-
-&cpu1 {
-	/delete-property/ cpu-idle-states;
-};
-
-&cpu2 {
-	/delete-property/ cpu-idle-states;
-};
-
-&cpu3 {
-	/delete-property/ cpu-idle-states;
-};
-
-/ {
-	cpus {
-		/delete-node/ idle-states;
-	};
-	/delete-node/ scpi;
-};
-
-&mbus{
-    status="disabled";
-};
-
-&mali {
-	assigned-clock-rates = <297000000>;
-};
