diff --git a/scripts/KlipperScreen-install.sh b/scripts/KlipperScreen-install.sh
index 53bf153..43899a1 100755
--- a/scripts/KlipperScreen-install.sh
+++ b/scripts/KlipperScreen-install.sh
@@ -117,7 +117,7 @@ create_virtualenv()
     if [ -d $KSENV ]; then
         echo_text "Removing old virtual environment"
         rm -rf ${KSENV}
-
+    fi
     echo_text "Creating virtual environment"
     python3 -m venv ${KSENV}
 
